plugins/inputs/mysql/mysql.go

```diff
@@ -641,11 +658,24 @@ func (m *Mysql) gatherSlaveStatuses(db *sql.DB, serv string, acc telegraf.Accumu
 			if m.MetricVersion >= 2 {
 				col = strings.ToLower(col)
 			}
-			if value, ok := m.parseValue(*vals[i].(*sql.RawBytes)); ok {
+
+			if m.GatherAllSlaveChannels &&
+				(strings.ToLower(col) == "channel_name" || strings.ToLower(col) == "connection_name") {
+				// Since the default channel name is empty, we need this block
+				channelName := "default"
+				if len(*vals[i].(*sql.RawBytes)) > 0 {
```

Comment:
Is `vals[i]` _always_ of type `*sql.RawBytes`? Otherwise this type assertion might panic...

Reply:
First of all, as you can see, I did not add this line. It was there from the beginning. Second, the definition of the type is "type RawBytes []byte" - any SQL response can be converted to this type, right?

Reply:
Ok, seems reasonable for me.